helper.py

```diff
@@ -1,41 +1,186 @@
+stat_file = img_static_path + "imgtime.txt"
 mod_time = 0
 out_width = 640.0
 out_height = 480.0
 
-while True:
-    time.sleep(0.1)
-    if os.path.exists(input_file):
-        stat_obj = os.stat(input_file)
-        if mod_time != stat_obj.st_mtime:
-            try:
-                img = Image.open(input_file)
-                width_i, height_i = img.size
-                width = width_i * 1.0
-                height = height_i * 1.0
-                scale_x = out_width / width
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+log_static_path = "/media/RobotUSB/"
+log_input_file = log_static_path + "logs.txt"
+
+picture_watcher.watch(
+    alias="image", path=img_input_file, flags=aionotify.Flags.MODIFY
+)  # sets up watcher
+log_watcher.watch(alias="logs", path=log_input_file, flags=aionotify.Flags.MODIFY)
+
+
+def shrink_image(img):
+    width_i, height_i = img.size
+    width = float(width_i)
+    height = float(height_i)
+    scale_x = out_width / width
+    scale_y = out_height / height
+    if scale_x > scale_y:
+        img = img.resize((int(width * scale_y), int(height * scale_y)), Image.LANCZOS)
+    else:
+        img = img.resize((int(width * scale_x), int(height * scale_x)), Image.LANCZOS)
+    return img
+
+
+def im_2_b64(image):
+    buff = BytesIO()
+    image.save(buff, format="JPEG")
+    img_str = base64.b64encode(buff.getvalue())
+    return img_str
+
+
+async def wait_for_picture_change():
+    loop = asyncio.get_event_loop()
+
+    bypass = False  # so first image is not ignored.
+    while not os.path.exists(img_input_file):
+        await asyncio.sleep(0.5)  # twiddle thumbs :)
+        if not bypass:
+            bypass = True
+    await picture_watcher.setup(loop)
+    print("Image change watcher is running.")
+
+    while True:  # for all events
+        if not bypass:
+            event = await picture_watcher.get_event()  # blocks until file changed
+        else:
+            bypass = False  # reset bypass
+
+        for c in range(10):
+            await asyncio.sleep(0.01)  # give it time to write the file.
+            try:  # this runs until the bot has finished writing the image
+                img = Image.open(img_input_file)
+                img.load()
+                print("Opened successfully")
+                break
+            except:
+                print("Error opening file: attempt \#" + str(c))
+
+        if c >= 9:
+            continue  # error with this file, go back and wait for next change.
```

Comment:
Maybe log this error as well? Like say something like reached max errors, just so you know it didn't succeed on the last try.

Reply:
The pythonic way of doing this is:
```py
for c in range(10):
     stuff()
else:
     we failed doing stuff
```

But I think your way is clearer to most people as its quite rare to see that syntax.